web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -230,6 +230,10 @@ const Resources = memo(
     const { fetch, resources, attempt, clear } = useUnifiedResourcesFetch({
       fetchFunc: useCallback(
         async (paginationParams, signal) => {
+          // Do not make the call if we don't know what resources we should show.
+          if (props.showResources === ShowResources.UNSPECIFIED) {
+            return { startKey: '', agents: [] };
+          }
```

Comment:
How does this work? Just by looking at this part of the code alone, it appears that it's wrong: if we get to this point before receiving cluster details, then the resources tab is going to flash "No resources" for a brief time before actually making a request.

To test this, I made the `GetCluster` RPC artificially slower by 3s on a local cluster. It seems to be handled correctly, the tab is stuck on the loading spinner.

<image alt="but why" src="https://storage.googleapis.com/sourcegraph-assets/blog/ryan-reynold-but-why-meme.jpg" width="250"/>

Reply:
Oh, I see, it's because of this:

https://github.com/gravitational/teleport/blob/f2907754c2630698617315b783269a9dfe0a6dcd/web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx#L325-L329

Reply:
I wonder if it wouldn't make more sense to make `fetchFunc` block forever until `showResources` are fetched? Otherwise the code that does the fetching is desynchronized with the code that reports the status of the fetching. This is what led to my original comment here in the first place.

We could have a function similar to wait, but it'd only take a signal and never resolve until the signal is aborted.

Reply:
In the first iteration I had something like `waitForever` but then I replaced it with that combined `resourcesFetchAttempt`. 
I'm happy to bring it back, it will make the code easier to understand. I had to adjust the code `useKeyBasedPagination.ts` so now it aborts the pending request when the hook unmounts.